packages/react-native-otlp/android/src/main/java/io/embrace/rnembraceotlp/RNEmbraceOTLPModule.kt

```diff
@@ -1,50 +1,177 @@
 package io.embrace.rnembraceotlp
 
-import android.util.Log
+import java.time.Duration
+import java.util.logging.Logger
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import kotlin.time.toJavaDuration
+
 import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
 import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableArray
 import com.facebook.react.bridge.ReadableMap
 
 import io.embrace.android.embracesdk.Embrace
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter
+
+data class HeaderConfig (
+    val key: String,
+    val token: String
+)
+
+data class ExporterConfig(
+    val endpoint: String,
+    val headers: List<HeaderConfig>? = emptyList(),
+    val timeout: Duration? = null
+)
+
+data class OtlpExporterConfig (
+    val traceExporter: ExporterConfig? = null,
+    val logExporter: ExporterConfig? = null
+)
+
+// parsing Readable Map into the real OTLP Exporter Config shape
+fun parseExportConfig(spanExportConfig: ReadableMap, logExportConfig: ReadableMap): OtlpExporterConfig {
+    var spanExportConfigParsed: ExporterConfig? = null
+    val spanExportEndpoint = spanExportConfig.getString("endpoint")
+
+    var logExportConfigParsed: ExporterConfig? = null
+    val logExportEndpoint = logExportConfig.getString("endpoint")
+
+    if (!spanExportEndpoint.isNullOrBlank()) {
+        spanExportConfigParsed = ExporterConfig(
+            endpoint = spanExportEndpoint,
+            headers = spanExportConfig.getArray("headers")?.let {
+                parseHeaders(it)
+            } ?: emptyList(),
+            timeout = parseTimeout(spanExportConfig.getDouble("timeout")),
+        )
+    }
+
+    if (!logExportEndpoint.isNullOrBlank()) {
+        logExportConfigParsed = ExporterConfig(
+            endpoint = logExportEndpoint,
+            headers = logExportConfig.getArray("headers")?.let {
+                parseHeaders(it)
+            } ?: emptyList(),
+            timeout = parseTimeout(logExportConfig.getDouble("timeout")),
+        )
+    }
+
+    return OtlpExporterConfig(traceExporter = spanExportConfigParsed, logExporter = logExportConfigParsed)
+}
+
+fun parseHeaders(headers: ReadableArray): List<HeaderConfig> {
+    val log = Logger.getLogger("[Embrace]")
```

Comment:
not sure how much overhead this involves but could also pass this in to the method to avoid having to re-setup

Reply:
done, moved methods into the class